app/models/components/course/materials_ability_component.rb

```diff
@@ -0,0 +1,6 @@
+module Course::MaterialsAbilityComponent
+  include AbilityHost::Component
+  extend ActiveSupport::Concern
+
+  include Course::Material::MaterialAbility
```

Comment:
If it's simple enough, you can. The assessments ability is extracted out because it is quite big.